docs/content/engines/spark3.md

```diff
@@ -160,52 +158,244 @@ INSERT INTO my_table VALUES (1, 'Hi'), (2, 'Hello');
 create table my_table (
     k int,
     v string
-) USING paimon tblproperties (
+) USING paimon
+tblproperties (
     'primary-key' = 'k'
 ) ;
 
-INSERT INTO my_table VALUES (1, 'Hi'), (2, 'Hello');
 ```
 
 {{< /tab >}}
 
 {{< /tabs >}}
 
-**Step 4: Query Table with SQL**
+## Insert Table
+
+```sql
+INSERT INTO my_table VALUES (1, 'Hi'), (2, 'Hello');
+```
+
+## Query Table
+
+{{< tabs "Query Paimon Table" >}}
+
+{{< tab "SQL" >}}
 
 ```sql
 SELECT * FROM my_table;
+
 /*
 1	Hi
 2	Hello
 */
 ```
 
-**Step 5: Update the Records**
+{{< /tab >}}
+
+{{< tab "DataFrame" >}}
+
+```scala
+val dataset = spark.read.format("paimon").load("file:/tmp/paimon/default.db/my_table")
+dataset.show()
+
+/*
++---+------+
+| k |     v|
++---+------+
+|  1|    Hi|
+|  2| Hello|
++---+------+
+*/
+```
+
+{{< /tab >}}
+
+{{< /tabs >}}
+
+
+## Update Table
+
+For now, Paimon does not support `UPDATE` syntax. But we can use `INSERT INTO` syntax instead for changelog tables.
 
 ```sql
 INSERT INTO my_table VALUES (1, 'Hi Again'), (3, 'Test');
 
 SELECT * FROM my_table;
+
 /*
 1	Hi Again
 2	Hello
 3	Test
 */
 ```
 
-**Step 6: Query Table with Scala API**
+## Streaming Write
 
-If you don't want to use Paimon catalog, you can also run `spark-shell` and query the table with Scala API.
+{{< hint info >}}
 
-```bash
-spark-shell ... --jars /path/to/paimon-spark-3.3-{{< version >}}.jar
+Paimon currently supports Spark 3+ for streaming write.
+
+Paimon Structured Streaming only supports the two `append` and `complete` modes.
+
+{{< /hint >}}
+
+```scala
+// Create a paimon table if not exists.
+spark.sql(s"""
+           |CREATE TABLE T (k INT, v STRING)
+           |TBLPROPERTIES ('primary-key'='a', 'write-mode'='change-log', 'bucket'='3')
+           |""".stripMargin)
+
+// Here we use MemoryStream to fake a streaming source.
+val inputData = MemoryStream[(Int, String)]
+val df = inputData.toDS().toDF("k", "v")
+
+// Streaming Write to paimon table.
+val stream = df
+  .writeStream
+  .outputMode("append")
+  .option("checkpointLocation", "/path/to/checkpoint")
+  .format("paimon")
+  .start("/path/to/paimon/sink/table")
 ```
 
+## Streaming Read
+
+{{< hint info >}}
+
+Paimon currently supports Spark 3.3+ for streaming read.
+
+{{< /hint >}}
+
+Paimon supports rich scan mode for streaming read. There is a list:
+<table class="configuration table table-bordered">
```

Comment:
Could we use the following style for configuration?

```
| Scan Mode  | Description                                                                  |
|-------------|--------------------------------------------------------|
| latest | For streaming sources, continuously reads latest changes without producing a snapshot at the beginning.|
```

Reply:
what's the difference. See ` Spark Type Conversion` use the table style.